flexget/plugins/operate/free_space.py

```diff
@@ -1,19 +1,59 @@
 import shutil
-from pathlib import Path
+from enum import Enum
+from pathlib import Path, PurePath
 
+from datamodel_code_generator.util import field_validator, model_validator
 from loguru import logger
+from pydantic import BaseModel, ByteSize, ConfigDict
 
 from flexget import plugin
-from flexget.config_schema import parse_size
 from flexget.event import event
 
 logger = logger.bind(name='free_space')
 
-ABORT_BELOW = 'below'
-ABORT_ABOVE = 'above'
 
-
-def get_free_space(config, task):
+class AbortMode(str, Enum):
+    BELOW = 'below'
+    ABOVE = 'above'
+
+
+class Config(BaseModel):
+    space: ByteSize | float
```

Comment:
Don't really love this ByteSize | float. The | float is just to generate the schema to allow floats from the user, but when we make an instance of this class it will always be a ByteSize